### PR TITLE
Incorrect parent

### PR DESCRIPTION
**Pull request by reynoldtan**
This PR proposes major changes to the code-base of this module. The main change being the query considering the implementation of a 'caretaker/sitter' in references to a stock. In addition, this pr eliminated redundant query (ie. sample shown in tool tip text) and created system variables for frequently used ontologies (ie. is_selection_of).

## Metadata

 - Issue https://github.com/UofS-Pulse-Binfo/germ_summary/issues/13 Incorrect summary count per parent cross.

- [X] I feel this PR is ready to be merged.
- [ ] This PR is dependent upon [PR #/ nothing]

Documentation:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Testing?
1. Disable and uninstall this module.
2. Update files.
3. Enable module.

I have verified summary count of crosses where it is less than a hundred. Unknown by Unknown and culinaris and culinaris both returned count more than 700+, I've looked at some items but I could not confirm nor verify its correctness. 

- [ ] I tested on a generic Tripal Site
- [X] I tested on a KnowPulse Clone
- [ ] This PR includes automated testing

### CODE REVIEW

**Comment by laceysanderson** [comment]
For more context when reviewing:
The broken version of the matrix looked like this:
Screen Shot 2019-09-19 at 2 24 15 PM
Notice the large number of `unknown x unknown`

The module has been updated on dev/fresh. Now we need to check that 
1. the counts are correct for each cell by cross referencing the RIL search
2. when a count is clicked on you see the crosses you expect based on known parentage.
3. generation information is filled in if available.
4. RIL name link goes to the germplasm page (i.e. LR-18)

**Comment by reynoldtan** [comment]
Confirmed! 
1. I've check the summary count matched the result in RIL search
2. Stocks the count was based on matched the count in the cell, as well as the stocks in the RIL search per cross.
3. Tested F2, F3, F4, F5, F6, F7, F8 and Rill Complete.
Could not find F1 field though and there's **F9?**. 
Screen Shot 2019-09-23 at 8 41 37 AM (2)

**Comment by ruobinLiu** [comment]
Checked, no issues was found.

File: germ_summary.module [hunk 1]
```diff
@@ -39,587 +39,431 @@ function germ_summary_menu() {
...
+
+          // Dad's name.
+          $form['siblings_dad'] = array(
+            '#type' => 'hidden',
+            '#value' => $args_dad,
+          );
+        }
+        else {
+          // 0 children for this pair.
+          drupal_set_message('Stocks not found.', 'warning');
```
Comment (laceysanderson) [inline germ_summary.module]:
```suggestion
          drupal_set_message('There are no RILs for this combination of species.', 'warning');
```

**Review by laceysanderson — CHANGES_REQUESTED** [review]
Links don't work for to go from matrix to listing for non-Lens matrix. For example, when I go to `germplasm/summary/Phaseolus` and click on `9` for `vulgaris x vulgaris` I am redirected to the Lens matrix...

Other then that, everything looks good :-)

**Review by laceysanderson — APPROVED** [review]

**Pull request merged**